### PR TITLE
Allow for annotating image-sequences when annotations not already present

### PR DESCRIPTION
**Pull request by AlmightyYakob**
This addresses the major concern raised by @mattdawkins. The diff is fairly large, as there needed to be several changes to get this to work, including fixing usability.

### CODE REVIEW

**Comment by subdavis** [comment]
This will mega-conflict with #17 

@BryonLewis was first, so the git gods command ye to rebase (after he merges).  :(

**Comment by AlmightyYakob** [comment]
This should be good regarding conflicts with #17. Since I've rebased off that branch, this PR should wait until that one is merged.

File: client/src/views/Viewer.vue [hunk 1]
```diff
@@ -411,18 +423,27 @@ export default {
...
-        return Object.freeze(detection);
-      });
+      let detections = [];
+      try {
+        const { data } = await this.girderRest.get("viame_detection", {
+          params: { folderId: this.dataset._id }
+        });
+
+        detections = data
+          ? data.map(detection => Object.freeze(detection))
```
Comment (subdavis) [inline client/src/views/Viewer.vue]: I know it was never commented before, but using object.freeze to protect objects from reactivity like this really deserves some explanation.  Some developer will come along later and wonder what was going on.

File: server/viame_server/event.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        Item().save(item)
+
+        folder = Folder().findOne({"_id": item["folderId"]})
+
+        # FPS is hardcoded for now
+        folder["meta"].update({"type": "image-sequence", "viame": True, "fps": 30})
+        Folder().save(folder)
+
+
+def mark_folder_for_annotation(event):
```
Comment (subdavis) [inline server/viame_server/event.py]: This function runs for all uploads, but only has an effect for folders.  You could rename this function `maybe_mark_folder_for_annotation(event)` to indicate that it doesn't always do that.

I think this function is straightforward, so you don't have to, just wanted to point out a pattern I've seen elsewhere in the girder ecosystem (particularly in worker)

File: client/src/views/Viewer.vue (hunk 1 above)
Comment (AlmightyYakob) [reply client/src/views/Viewer.vue]: I'm not sure why it's being done either, it was there before I came onto the project. I think it's deserving of it's own PR though.

**Comment by AlmightyYakob** [comment]
@subdavis I think I addressed your concerns. I also added a small change in 
f421a26 to match the client's [list](https://github.com/VIAME/VIAME-Web/blob/435fd702e322b7aefa0763b3b31464474449002e/client/src/views/Viewer.vue#L97-L99) of valid image formats, tested locally no problem.

File: client/src/views/Viewer.vue [hunk 3]
```diff
@@ -422,6 +422,11 @@ export default {
     /* END TODO */
     async loadDataset(datasetId) {
       var { data: dataset } = await this.girderRest.get(`folder/${datasetId}`);
+
+      if (!dataset) {
+        throw "Could not fetch dataset!";
```
Comment (subdavis) [inline client/src/views/Viewer.vue]: I've actually never seen anyone do this.  I found an old stackoverflow thread that says this behaves unexpectedly in IE and firefox, but that was in 2012.  My gut says `throw new Error()` should always be used, but maybe you know something I don't.

**Pull request merged**